grype/db/v6/vulnerability.go

```diff
@@ -0,0 +1,240 @@
+package v6
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/anchore/grype/grype/pkg/qualifier"
+	"github.com/anchore/grype/grype/pkg/qualifier/platformcpe"
+	"github.com/anchore/grype/grype/pkg/qualifier/rpmmodularity"
+	"github.com/anchore/grype/grype/version"
+	"github.com/anchore/grype/grype/vulnerability"
+	"github.com/anchore/syft/syft/cpe"
+)
+
+func newVulnerabilityFromAffectedPackageHandle(rdr Reader, affected AffectedPackageHandle, affectedRange AffectedRange) (*vulnerability.Vulnerability, error) {
+	packageName := ""
+	if affected.Package != nil {
+		packageName = affected.Package.Name
+	}
+
+	vuln := affected.Vulnerability
+	if vuln == nil || vuln.BlobValue == nil {
+		var err error
+		vuln, err = getVulnerability(rdr, affected.VulnerabilityID)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return newVulnerabilityFromParts(packageName, vuln, affected.BlobValue, affectedRange, &affected, nil)
+}
+
+func newVulnerabilityFromAffectedCPEHandle(rdr Reader, affected AffectedCPEHandle, affectedRange AffectedRange) (*vulnerability.Vulnerability, error) {
+	vuln := affected.Vulnerability
+	if vuln == nil || vuln.BlobValue == nil {
+		var err error
+		vuln, err = getVulnerability(rdr, affected.VulnerabilityID)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return newVulnerabilityFromParts(affected.CPE.Product, vuln, affected.BlobValue, affectedRange, nil, &affected)
+}
+
+func getVulnerability(rdr Reader, id ID) (*VulnerabilityHandle, error) {
+	var err error
+	vulns, err := rdr.GetVulnerabilities(&VulnerabilitySpecifier{ID: id}, &GetVulnerabilityOptions{
+		Preload: true,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(vulns) == 0 {
+		return nil, fmt.Errorf("no vulnerabilities found for package %v", id)
+	}
+	return &vulns[0], nil
+}
+
+func newVulnerabilityFromParts(packageName string, vuln *VulnerabilityHandle, affected *AffectedPackageBlob, affectedRange AffectedRange, affectedPackageHandle *AffectedPackageHandle, affectedCpeHandle *AffectedCPEHandle) (*vulnerability.Vulnerability, error) {
+	if vuln.BlobValue == nil {
+		return nil, fmt.Errorf("vuln has no blob value: %+v", vuln) // FIXME we could just query the db
```

Comment:
```suggestion
		return nil, fmt.Errorf("vuln has no blob value: %+v", vuln)
```
within the constructor there really shouldn't be any DB access